packages/host/app/components/operator-mode/code-submode/spec-preview.gts

```diff
@@ -477,7 +512,23 @@ export default class SpecPreview extends GlimmerComponent<Signature> {
   @service private declare recentFilesService: RecentFilesService;
   @service private declare specPanelService: SpecPanelService;
   @service private declare store: StoreService;
-  @tracked private search: ReturnType<getCards<Spec>> | undefined;
+  @tracked cards: Spec[] = [];
+
+  constructor(owner: Owner, args: any) {
+    super(owner, args);
+    this.search.perform();
+  }
+
+  private search = task(async () => {
+    this.cards = flatMap(
+      await Promise.all(
+        this.realms.map(
+          async (realm) =>
+            await this.store.search(this.specQuery, new URL(realm)),
+        ),
+      ),
+    ) as Spec[];
+  });
```

Comment:
Ahh I can see how this search will not re-run from choosing a different code ref in lhs selector. Hence, why we may want a resource. Perhaps, there is an intent to maintain `getCards`

Reply:
so yeah, i think `getCards()` is what you want here right? the search query is a thunk than when it is changed the search re-runs. `Store.search()` is just a straight-up one-shot search.

Reply:
the way you are consuming the search results, though is going to cause flickering, as you are just returning an array. the array is probably different each time the search is run--there is no object stability. we solved in this other places by actually passing the _whole resource_ to the child component (in the ai assistant). the search resource is actually stable between different searches and will not trigger a re-render because of object equality

Reply:
in general, when you see an array being specified in a component signature, alarm bells should go off. arrays are notoriously unstable and can cause flickering if we are replacing arrays instead of mutating them.